core/src/Kokkos_Parallel.hpp

```diff
@@ -175,14 +175,16 @@ template <class ExecPolicy, class FunctorType>
 inline void parallel_for(const std::string& str, const ExecPolicy& policy,
                          const FunctorType& functor) {
 #if KOKKOS_ENABLE_DEBUG_PRINT_KERNEL_NAMES
-  Kokkos::fence();
+  Kokkos::fence(
+      "Kokkos::parallel_for: debug fence printing kernel names before launch");
   std::cout << "KOKKOS_DEBUG Start parallel_for kernel: " << str << std::endl;
 #endif
 
   ::Kokkos::parallel_for(policy, functor, str);
 
 #if KOKKOS_ENABLE_DEBUG_PRINT_KERNEL_NAMES
```

Comment:
Looks like this is never defined...

Reply:
@crtrott can correct me, I believe this is an advanced feature, only set in the CMAKE_CXX_FLAGS, there's no corresponding CMake option

Reply:
Yes, that's what I meant. Obviously, not related to this pull request. Maybe, we discuss somewhere else if we should document this better.

Reply:
Of course, such users *should* be using the Kokkos Tools subsystem, writen by that smrt guy, for these porpoises

Reply:
See https://github.com/kokkos/kokkos/pull/4150 for discussing if/what we want to do about it.